Modules/mmapmodule.c

```diff
@@ -292,7 +292,7 @@ filter_page_exception_method(mmap_object *self, EXCEPTION_POINTERS *ptrs,
 }
 #endif
 
-#if defined(MS_WINDOWS) && !defined(DONT_USE_SEH)
+#if (defined(MS_WINDOWS_DESKTOP) || defined(MS_WINDOWS_SYSTEM)) && !defined(DONT_USE_SEH)
```

Comment:
Can we maybe rephrase as something like this at the top of the file? We have similar code over in `ctypes/callproc.c`, and I think it'll make it more likely that if/when we add more SEH handling, we'll get it right. (Update the comment as appropriate, I just made up something feasible, not necessarily correct)

```
#if !defined(DONT_USE_SEH) && !(defined(MS_WINDOWS_DESKTOP) || defined(MS_WINDOWS_SYSTEM))
// Only the WINDOWS_DESKTOP and WINDOWS_SYSTEM API partitions support SEH handling
// the way we want to use it.
#define DONT_USE_SEH
#endif
```